modules/reactor/src/meshgenerators/ControlDrumMeshGenerator.C

```diff
@@ -0,0 +1,509 @@
+                       "axial zone (drum inner, drum, and drum outer)";
+      paramError("region_ids", err_msg);
+    }
+  }
+  else
+    mooseError(
+        "Region IDs must be assigned for ControlDrumMeshGenerator using parameter 'region_ids'");
+
+  // Check block names have the correct size
+  if (isParamValid("block_names"))
+  {
+    if (getReactorParam<bool>(RGMB::region_id_as_block_name))
+      paramError("block_names",
+                 "If ReactorMeshParams/region_id_as_block_name is set, block_names should not be "
+                 "specified in ControlDrumMeshGenerator");
+    _has_block_names = true;
+    _block_names = getParam<std::vector<std::vector<std::string>>>("block_names");
+    if (_region_ids.size() != _block_names.size())
+      mooseError("The size of block_names must match the size of region_ids");
+    for (const auto i : index_range(_region_ids))
+      if (_region_ids[i].size() != _block_names[i].size())
+        mooseError("The size of block_names must match the size of region_ids");
+  }
+  else
+    _has_block_names = false;
+
+  // Check extrusion parameters
+  if (_extrude && _mesh_dimensions != 3)
+    paramError("extrude",
+               "This is a 2 dimensional mesh, you cannot extrude it. Check your ReactorMeshParams "
+               "inputs\n");
+  if (_extrude && (!hasReactorParam<boundary_id_type>(RGMB::top_boundary_id) ||
+                   !hasReactorParam<boundary_id_type>(RGMB::bottom_boundary_id)))
+    mooseError("Both top_boundary_id and bottom_boundary_id must be provided in ReactorMeshParams "
+               "if using extruded geometry");
+
+  // No subgenerators will be called if option to bypass mesh generators is enabled
+  if (!getReactorParam<bool>(RGMB::bypass_meshgen))
+  {
+    const std::string block_name_prefix =
+        RGMB::DRUM_BLOCK_NAME_PREFIX + std::to_string(_assembly_type);
+
+    // Define azimuthal angles explicitly based on num_azimuthal_sectors and manually add pad start
+    // angle and end angle if they are not contained within these angle intervals
+    std::vector<Real> azimuthal_angles;
+    const auto custom_start_angle = _pad_start_angle;
+    const auto custom_end_angle = (_pad_end_angle > 360) ? _pad_end_angle - 360. : _pad_end_angle;
+    for (unsigned int i = 0; i < num_sectors; ++i)
+    {
+      Real current_azim_angle = (Real)i * 360. / (Real)num_sectors;
+      Real next_azim_angle = (Real)(i + 1) * 360. / (Real)num_sectors;
+      azimuthal_angles.push_back(current_azim_angle);
+
+      if (!MooseUtils::absoluteFuzzyEqual(current_azim_angle, custom_start_angle) &&
+          !MooseUtils::absoluteFuzzyEqual(next_azim_angle, custom_start_angle) &&
+          (custom_start_angle > current_azim_angle) && (custom_start_angle < next_azim_angle))
+      {
+        mooseWarning("pad_start_angle not contained within drum azimuthal discretization so "
+                     "additional azimuthal nodes are created to align mesh with this angle");
```

Comment:
How are you imagining this tolerance parameter be defined? Would it be an absolute value (radians) or in terms of degrees? I personally think adding a parameter here to control tolerance would lead to added complexity / source of confusion for the user and would prefer to just add that thin azimuthal interval in that case

Reply:
IMO it should be an absolute value (rad or deg). The current code would create nearly zero volume elements. At least use something like `absoluteFuzzyGreaterThan` to avert that (still need a tolerance). Moving nodes would make the element quality better but this might not be a hard requirement.

Reply:
I've added a parameter `azimuthal_node_tolerance` that is set to 0.1 degrees by default to accomplish this, let me know if this works